src/utils/size-observer.js

```diff
@@ -0,0 +1,142 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {pushIfNotExist, removeItem} from './array';
+import {rethrowAsync} from '../log';
+
+/** @typedef {function(../layout-rect.LayoutSizeDef)} */
+let ContentSizeObserverCallbackDef;
+
+/** @const {!WeakMap<!Window, !ResizeObserver>} */
+const observers = new WeakMap();
+
+/** @const {!WeakMap<!Element, !ContentSizeObserverCallbackDef>} */
+const targetObserverListMap = new WeakMap();
+
+/** @const {!WeakMap<!Element, !../layout-rect.LayoutSizeDef>} */
+const targetSizeListMap = new WeakMap();
+
+/**
+ * @param {!Element} element
+ * @param {!ContentSizeObserverCallbackDef} callback
+ */
+export function observeContentSize(element, callback) {
+  const win = element.ownerDocument.defaultView;
```

Comment:
should this be a `devAssert` instead? we shouldn't be observing detached elements

Reply:
It's not so much for detached elements. And in dev mode (where `devAssert` would work) this situation would never happen. This is only a protection for FIE that often cause some spurious errors like these when an iframe is removed. And this is to avoid overspamming ourselves. I can remove for now, but I added this initially because this code explicitly deals with multi-window case.